yarn-project/aztec.js/src/contract/contract_function_interaction.ts

```diff
@@ -63,18 +64,48 @@ export class ContractFunctionInteraction extends BaseContractInteraction {
   }
 
   /**
-   * Execute a view (read-only) transaction on an unconstrained function.
-   * This method is used to call functions that do not modify the contract state and only return data.
-   * Throws an error if called on a non-unconstrained function.
+   * Simulate a transaction and get its return values
+   * Differs from prove in a few important ways:
+   * 1. It returns the values of the function execution
+   * 2. It supports `unconstrained`, `private` and `public` functions
+   * 3. For `private` execution it:
+   * 3.a SKIPS the entrypoint and starts directly at the function
+   * 3.b SKIPS public execution entirely
+   * 4. For `public` execution it:
+   * 4.a Removes the `txRequest` value after ended simulation
+   * 4.b Ignores the `from` in the options
+   *
    * @param options - An optional object containing additional configuration for the transaction.
-   * @returns The result of the view transaction as returned by the contract function.
+   * @returns The result of the transaction as returned by the contract function.
    */
-  public view(options: ViewMethodOptions = {}) {
-    if (this.functionDao.functionType !== FunctionType.UNCONSTRAINED) {
-      throw new Error('Can only call `view` on an unconstrained function.');
+  public async simulate(options: SimulateMethodOptions = {}): Promise<any> {
+    if (this.functionDao.functionType == FunctionType.UNCONSTRAINED) {
+      return this.wallet.viewTx(this.functionDao.name, this.args, this.contractAddress, options.from);
     }
 
-    const { from } = options;
-    return this.wallet.viewTx(this.functionDao.name, this.args, this.contractAddress, from);
+    // TODO: If not unconstrained, we return a size 4 array of fields.
+    // TODO: It should instead return the correctly decoded value
+    // TODO: The return type here needs to be fixed! @LHerskind
+
+    if (this.functionDao.functionType == FunctionType.SECRET) {
```

Comment:
Having `FunctionType.SECRET` for private functions is confusing and it made it harder to decode what's going on here. Do you know if there is some justification for why that is the case or if it's just a tech debt?

Reply:
I think this is just straight up horrible and old tech debt.